### PR TITLE
daemon: run commands through LE

### PR DESCRIPTION
**Pull request by marko-gacesa**

### CODE REVIEW

File: engine/compiler/compiler.go [hunk 1]
```diff
@@ -46,44 +48,60 @@ func (c *Compiler) Compile(ctx context.Context, args runtime.CompilerArgs) runti
...
+	pipelineOS := pool.GetOS()
+	pipelineRoot := pool.GetRootDir()
+
 	// move the pool from the `mapping of pools` into the spec of this pipeline.
 	spec.CloudInstance.PoolName = targetPool
-	spec.Root = pool.GetRootDir()
 
-	pipelineOS := pool.GetOS()
 	// creates a home directory in the root.
 	// note: mkdirall fails on windows so we need to create all directories in the tree.
```
Comment (shubham149) [inline engine/compiler/compiler.go:63]: In golang, os.MkdirAll works for both windows & linux I guess.

File: engine/compiler/compiler.go [hunk 2]
```diff
@@ -163,22 +183,35 @@ func (c *Compiler) Compile(ctx context.Context, args runtime.CompilerArgs) runti
...
+
+		// TODO: Remove when lespec.Step is with Files slice
+		_ = clonepath
+		_ = clonefile
+
 		cmd, args := oshelp.GetCommand(pipelineOS, clonepath)
 		spec.Steps = append(spec.Steps, &engine.Step{
-			Name:      "clone",
+			Step: lespec.Step{
+				ID:   random(),
```
Comment (shubham149) [inline engine/compiler/compiler.go:194]: Also, set a Key field in all the steps. It can be same as ID.

Comment (marko-gacesa) [reply engine/compiler/compiler.go:194]: What is the exact name of the field? I can't find it...

Comment (shubham149) [reply engine/compiler/compiler.go:194]: https://github.com/harness/lite-engine/blob/main/api/api.go#L47

Comment (shubham149) [reply engine/compiler/compiler.go:194]: LogKey

**Review by tphoney — APPROVED** [review]

**Pull request merged**